pandas/tests/groupby/test_categorical.py

```diff
@@ -963,3 +964,33 @@ def test_shift(fill_value):
                               categories=['a', 'b', 'c', 'd'], ordered=False)
     res = ct.shift(1, fill_value=fill_value)
     assert_equal(res, expected)
+
+
+@pytest.mark.parametrize("observed, index, op, data", [
+    (True, 'multi_index_cat_partial', 'agg', [3, 3, 4]),
```

Comment:
You can add a separate decorator for `op` parametrized on agg/apply rather than having to duplicate each time here

Reply:
Could also use the observed fixture separately

Reply:
I think different values for index prevent me from doing that: I don't want `apply` to run with index `multi_index_cat_partial` (line 970), and I don't want to run `agg` on `multi_index_non_cat_partial` (line 971). BaseGrouper.apply changes the index when `observed=True` and I did not find a simple way to keep it, thus the differences in indices. Looking at values only would make this more simple, but that does not feel sufficient.

The same goes for `observed`, as `observed=True` and `observed=False` return different indices, I can't run all values of `observed` against all values of expected indices.

Reply:
Oh OK I misread those values. You can also split those out into a separate test if they should exhibit different behavior